src/Assets/LightSkyDefense/Scripts/GameManager.cs

```diff
@@ -4,16 +4,84 @@
 
 public class GameManager : MonoBehaviour
 {
-    // Start is called before the first frame update
-    void Start()
+    [HideInInspector]
+    public GameObject _wayPointPrefab;
+
+    private Vector3[] _calculatedPathPoints = null;
+
+    private static GameManager _instance;
+    public static GameManager Instance
```

Comment:
It's not necessary to make this object a singleton, the game manager will be part of the Player object. the Player object is already singleton. We can just use `Player.instance.GetComponent<GameManager>()` to get the GameManager instance.